meta-lmp-base/classes/go-mod-network.bbclass

```diff
@@ -0,0 +1,17 @@
+# Foundries LmP Go Modules with network support
+#
+# In LmP layer we need network support to build Go Modules
```

Comment:
Do we need to call the class lmp-go-mod-network? (only to make sure it's from lmp when inherit it from other files)

Reply:
yeah we can do it, I will change the name after when https://github.com/foundriesio/lmp-manifest/pull/221 finish.

Reply:
thinking about it again, this class is like a wrapper around the upstream version go-mod.bbclass, the main ideia is to don't lost changes from the upstream version.
It is not very easy to see the connection between the two prefixing with lmp `go-mod -> lmp-go-mod-network`. I think is better to have this without lmp in the name because the class works with all go modules and not only the ones in meta-lmp, ate the end it is a ugly hack to only enable network during compile.